Microsoft.NET.Build.Containers/AmazonECRMessageHandler.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.NET.Build.Containers;
+
+/// <summary>
+/// A delegating handler that handles the special error handling needed for Amazon ECR.
+/// 
+/// When pushing images to ECR if the target container repository does not exist ECR ends
+/// the connection causing an IOException with a generic "The response ended prematurely."
+/// error message. The handler catches the generic error and provides a more informed error
+/// message to let the user know they need to create the repository.
+/// </summary>
+public class AmazonECRMessageHandler : DelegatingHandler
+{
+    public AmazonECRMessageHandler(HttpMessageHandler innerHandler) : base(innerHandler) { }
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        try
+        {
+            return await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
+        }
+        catch (HttpRequestException e) when (e.InnerException is IOException ioe && ioe.Message.Equals("The response ended prematurely.", StringComparison.OrdinalIgnoreCase))
```

Comment:
Is the string always the same, even on other locales?

Reply:
I worried about the same thing. The `The response ended prematurely` message is not coming from ECR because it just quits the request it is coming from the HttpClient. I tested setting my culture to `de-DE` and the exception message didn't change. I wish there was a better way to detect the error but I haven't found any. Worse case if this checks fails is the user gets the generic error message without giving the user the clue that they need to make sure the repository is created.

Reply:
Digging into this with the runtime team - I found the exception message, and it is a localizable resource string, but I can't find any other localizations.

Reply:
A more structured way of checking for this error is planned for .NET 8: https://github.com/dotnet/runtime/issues/76644

So for now this comparison should be fine, though we'll keep an eye on that issue and consider changing to use it when the API is available.